test/dynamo/test_export.py

```diff
@@ -2448,22 +2448,29 @@ def true_fn(x):
         def false_fn(x):
             return x.sin()
 
-        def f_pred_traced_as_constant_var(x):
-            return cond(x.dim() > 2, true_fn, false_fn, [x])
-
         def f_pred_traced_as_symnode_var(x):
-            return cond(x.shape[0] > 10, true_fn, false_fn, [x])
+            return cond(x.shape[0] > 2, true_fn, false_fn, [x])
 
         def f_pred_traced_as_tensor_var(x):
             return cond(x.all(), true_fn, false_fn, [x])
 
-        example_inputs = (torch.rand(5),)
+        def f_pred_complex_expression_traced_as_symnode_var(x):
+            return cond(
+                x.dim() > 1 and x.shape[1] > 5 and x.shape[1] <= 10,
```

Comment:
@tugsbayasgalan It turns out this is just traced fine:
```
test_cond_supported_pred_types_dynamic_shapes (torch._dynamo.testing.DynamicShapesExportTests) ...


def forward(self, x):
    arg0, = fx_pytree.tree_flatten_spec(([x], {}), self._in_spec)
    sym_size = torch.ops.aten.sym_size(arg0, 1)
    le = sym_size <= 10;  sym_size = None
    true_graph_0 = self.true_graph_0
    false_graph_0 = self.false_graph_0
    cond = torch.ops.cond(le, true_graph_0, false_graph_0, [arg0]);  le = true_graph_0 = false_graph_0 = arg0 = None
    return pytree.tree_unflatten([cond], self._out_spec)
```

Reply:
@tugsbayasgalan Maybe not completely correct. It looks like the condition is over simplified. But anyway the issue of predicate became specialized doesn't seem to be relevant to #98900. And I do see more similar cases in the exportdb when the predicate is simply a shape check (https://www.internalfb.com/intern/staticdocs/exportdb/cond_branch_nested_function.html). CC: @angelayi